salt/modules/aptpkg.py

```diff
@@ -175,13 +173,12 @@ def _parse_sources(self, line):
             if repo_line[0] not in ["deb", "deb-src", "rpm", "rpm-src"]:
                 self.invalid = True
                 return False
-            self.architectures = []
             if repo_line[1].startswith("["):
                 opts = re.search(r"\[.*\]", self.line).group(0).strip("[]")
-                repo_line = [x for x in repo_line if x.strip("[]")]
+                repo_line = [x.strip("[]") for x in repo_line if x.strip("[]")]
```

Comment:
**question** would `[x for x in (line.strip("[]") for line in repo_line) if x]` read clearer? Would eliminate the double `strip` :thinking: